pyanaconda/payload/livepayload.py

```diff
@@ -167,14 +167,25 @@ def install(self):
         threadMgr.wait(THREAD_LIVE_PROGRESS)
 
         # Live needs to create the rescue image before bootloader is written
-        if not os.path.exists(util.getSysroot() + "/usr/sbin/new-kernel-pkg"):
-            log.error("new-kernel-pkg does not exist - grubby wasn't installed?  skipping")
-            return
+        if os.path.exists(util.getSysroot() + "/usr/sbin/new-kernel-pkg"):
+            useNKP = True
+        else:
+            log.warning("new-kernel-pkg does not exist - grubby wasn't installed?")
+            useNKP = False
 
         for kernel in self.kernelVersionList:
             log.info("Generating rescue image for %s", kernel)
-            util.execInSysroot("new-kernel-pkg",
-                               ["--rpmposttrans", kernel])
+            if useNKP:
+                util.execInSysroot("new-kernel-pkg",
+                                   ["--rpmposttrans", kernel])
+            else:
+                files = glob.glob(util.getSysroot() + "/etc/kernel/postinst.d/*")
```

Comment:
Implementing the gathering of `postinst.d` plugins and executing them from here feels wrong — it feels like there should be some API or tools to do that. Not sure about this though, so I'd welcome other people to comment on this.